sample/src_polymer/elements/scene-perception-pages/main.js

```diff
@@ -5,38 +5,43 @@ const SP_SIZE_FPS = 60;
 const CONTROL_PANEL_WIDTH = 125;
 var myStatus = new Status();
 
-function initBindings(spDom) {
-  // SP status:
-  // 'idle'(0) - before init successfully,
-  // 'checking'(1) - init successfully,
-  // 'tracking'(2) - SP started
-  spDom.spState = 0;
-  spDom.extendReconstruction = true;
+function destroySP(spDom) {
+  var sp = getSP();
+  if (!sp || spDom.spState == 0) return;
+
+  sp.destroy().then(function() {
+    spDom.spState = 0;
+    myStatus.info('destroy succeeds');
+  }, errorHandler);
 }
 
 function main(spDom) {
-  // Global variables.
-  var sp;
-  var imagePanelWidth = SP_SIZE_WIDTH;
-  var imagePanelHeight = SP_SIZE_HEIGHT;
+  var sp = getSP();
+  if (!sp) return;
 
-  initBindings(spDom);
-
-  function initUI() {
+  if (!spDom._initialized) {
     spDom.$$('#bottomPanel').appendChild(myStatus.getDom());
-    spDom.$$('#rightHint').fitInfo = spDom.$$('#rightContainer');
-    spDom.$$('#leftHint').fitInfo = spDom.$$('#leftContainer');
-
-    resizeUI(null);
-  }
 
-  function destroySP() {
-    sp.destroy().then(function() {
-      spDom.spState = 0;
-      myStatus.info('destroy succeeds');
-    }, errorHandler);
+    var leftView = new LeftRender(sp, spDom);
+    var rightView = new RightRender(sp, spDom);
+    bindHandlers(leftView, rightView);
+    // SP status:
+    // 'idle'(0) - before init successfully,
+    // 'checking'(1) - init successfully,
+    // 'tracking'(2) - SP started
+    spDom.spState = 0;
   }
+  myStatus.info('Please wait for initialization.');
+  initSP(function() {
+    if (!spDom._initialized) {
+      leftView.init();
+      rightView.init();
+      spDom._initialized = true;
```

Comment:
No any code resetting `_initialized` to `false`? And, as we already have `spDom.spState`, maybe `_initialized` is redundant flag?